aws/ec2.go

```diff
@@ -0,0 +1,226 @@
+package aws
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/ec2metadata"
+	awssession "github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/ec2/ec2iface"
+	log "github.com/sirupsen/logrus"
+
+	apiv3 "github.com/projectcalico/libcalico-go/lib/apis/v3"
+)
+
+const (
+	timeout         = 20
+	retries         = 3
+	deviceIndexZero = 0
+)
+
+func convertError(err error) string {
+	if awsErr, ok := err.(awserr.Error); ok {
+		return fmt.Sprintf("%s: %s", awsErr.Code(), awsErr.Message())
+	}
+
+	return fmt.Sprintf("%v", err.Error())
+}
+
+func retriable(err error) bool {
+	if awsErr, ok := err.(awserr.Error); ok {
+		switch awsErr.Code() {
+		case "InternalError":
+			return true
+		case "InternalFailure":
+			return true
+		case "RequestLimitExceeded":
+			return true
+		case "ServiceUnavailable":
+			return true
+		case "Unavailable":
+			return true
+		}
+	}
+
+	return false
+}
+
+func checkSourceDestinationValueIsDisable(check string) bool {
+	return check == apiv3.AWSSrcDstCheckOptionDisable
+}
+
+func UpdateSrcDstCheck(check string) error {
+	ctx, cancel := context.WithTimeout(context.Background(), timeout*time.Second)
+	defer cancel()
+
+	ec2Cli, err := newEC2Client(ctx)
+	if err != nil {
+		return err
+	}
+
+	ec2NetId, err := ec2Cli.getEC2NetworkInterfaceId(ctx)
+	if err != nil {
+		return fmt.Errorf("error getting ec2 network-interface-id: %s", convertError(err))
+	}
+
+	checkVal := !checkSourceDestinationValueIsDisable(check)
```

Comment:
This line is confusing because it has ended up a double negative `not(is _dis_abled)`.  Why not just write it as `checkVal := check == apiv3.AWSSrcDstCheckOptionEnabled`

Given this is dealing with booleans and awkward naming from  AWS, I'd probably use more verbose variable name here:
```
s/check/caliCheckAction/
s/checkVal/checkEnabled/
```